homeassistant/components/ambee/sensor.py

```diff
@@ -1,69 +1,91 @@
     CoordinatorEntity,
     DataUpdateCoordinator,
 )
 
-from .const import DOMAIN, SENSORS
+from .const import ATTR_ENABLED_BY_DEFAULT, DOMAIN, SENSORS, SERVICES
+from .models import AmbeeSensor
 
 
 async def async_setup_entry(
     hass: HomeAssistant,
     entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Set up Ambee sensor based on a config entry."""
-    coordinator: DataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    """Set up Ambee sensors based on a config entry."""
     async_add_entities(
-        AmbeeSensor(coordinator=coordinator, entry_id=entry.entry_id, key=sensor)
-        for sensor in SENSORS
+        AmbeeSensorEntity(
+            coordinator=hass.data[DOMAIN][entry.entry_id][service_key],
+            entry_id=entry.entry_id,
+            sensor_key=sensor_key,
+            sensor=sensor,
+            service_key=service_key,
+            service=SERVICES[service_key],
+        )
+        for service_key, service_sensors in SENSORS.items()
+        for sensor_key, sensor in service_sensors.items()
     )
 
 
-class AmbeeSensor(CoordinatorEntity, SensorEntity):
+class AmbeeSensorEntity(CoordinatorEntity, SensorEntity):
     """Defines an Ambee sensor."""
 
     def __init__(
-        self, *, coordinator: DataUpdateCoordinator, entry_id: str, key: str
+        self,
+        *,
+        coordinator: DataUpdateCoordinator,
+        entry_id: str,
+        sensor_key: str,
+        sensor: AmbeeSensor,
+        service_key: str,
+        service: str,
     ) -> None:
         """Initialize Ambee sensor."""
         super().__init__(coordinator=coordinator)
-        self._key = key
-        self._entry_id = entry_id
-        self._service_key, self._service_name = SENSORS[key][ATTR_SERVICE]
+        self._sensor_key = sensor_key
+        self._service_key = service_key
+
+        self.entity_id = f"{SENSOR_DOMAIN}.{service_key}_{sensor_key}"
+        self._attr_device_class = sensor.get(ATTR_DEVICE_CLASS)
+        self._attr_entity_registry_enabled_default = sensor.get(
+            ATTR_ENABLED_BY_DEFAULT, True
+        )
+        self._attr_icon = sensor.get(ATTR_ICON)
+        self._attr_name = sensor.get(ATTR_NAME)
+        self._attr_state_class = sensor.get(ATTR_STATE_CLASS)
+        self._attr_unique_id = f"{entry_id}_{service_key}_{sensor_key}"
+        self._attr_unit_of_measurement = sensor.get(ATTR_UNIT_OF_MEASUREMENT)
 
-        self._attr_device_class = SENSORS[key].get(ATTR_DEVICE_CLASS)
-        self._attr_name = SENSORS[key][ATTR_NAME]
-        self._attr_state_class = SENSORS[key].get(ATTR_STATE_CLASS)
-        self._attr_unique_id = f"{entry_id}_{key}"
-        self._attr_unit_of_measurement = SENSORS[key].get(ATTR_UNIT_OF_MEASUREMENT)
+        self._attr_device_info = {
+            ATTR_IDENTIFIERS: {(DOMAIN, f"{entry_id}_{service_key}")},
+            ATTR_NAME: service,
+            ATTR_MANUFACTURER: "Ambee",
+        }
```

Comment:
Should we set `"entry_type"` to `"service"`?

Reply:
Well spotted 👍 

![image](https://user-images.githubusercontent.com/195327/121519241-c5564a00-c9f1-11eb-88f5-0502b0e2b825.png)